docs/src/BrandingProvider.tsx

```diff
@@ -17,14 +17,8 @@ export default function BrandingProvider(props: BrandingProviderProps) {
   const { children, mode: modeProp } = props;
   const upperTheme = useTheme();
   const mode = modeProp || upperTheme.palette.mode;
-  const theme = React.useMemo(
-    () =>
-      createTheme({
-        ...getDesignTokens(mode),
-        ...getThemedComponents(),
-      }),
-    [mode],
-  );
+  const theme = mode === 'dark' ? brandingDarkTheme : brandingLightTheme;
```

Comment:
No need to create the same theme over and over again between each `BrandingProvider` instance. In the Joy UI pages, we mount this component a lot of times.

---

**Off-topic** it seems that there are way too many ThemeProvider used in the pages, but it's a different problem, so I'm not looking into it. I think that what we need is:

1. One Material UI customized branding theme provider at the root of the page (not inside `_app`)
2. One raw Material UI theme provider for each Material UI demos, one raw Joy UI theme provider for each Joy UI demo, etc.